### ISSUE TITLE
What information shall we collect from the registration form?

### ISSUE THREAD

**Issue by Alan32Liu** [mboehme/FuzzingCompetition#2]
Currently, I can think of the following:
1. Fuzzer name;
2. The link to their GitHub PR;
3. Author(s) and their organisation(s);
5. Primary contact;
6. A [short technical report](https://github.com/mboehme/FuzzingCompetition/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R35);

Would there be any other information the committee would like to collect?
Thanks : )

**Comment by mboehme** [comment]
> 2. The link to their GitHub PR;

Do you want them to register only on the day of the competition? Or should we rather ask them to register beforehand?

Everything else looks good to me!

**Comment by Alan32Liu** [comment]
IMO, asking them to register beforehand is better, as it gives us more time to estimate the number of participants [we'd like to accept](https://github.com/mboehme/FuzzingCompetition/pull/1#discussion_r987484680) and get prepared. For example, they can submit the registration form once they submit a PR and finish the short report.
Which one would you prefer?

**Comment by mboehme** [comment]
Not sure I understand. They'll submit the PR only on the day of the competition, no?

**Comment by mboehme** [comment]
I'd prefer registering beforehand, too.

**Comment by Alan32Liu** [comment]
Ah, I reckon the day of the competition is the due day for their submission?
In that case, they will probably _submit_ their PRs before that day, as they will need time to fix CI build failures and ask questions.
We will _approve_ their PRs on the day of the competition and merge them, after which they will not be able to change their fuzzers, and then we will start running the final evaluation.

Would that sound good to you?
Hope I did not miss/misunderstand anything. More than happy to adapt to your plan : )

**Comment by mboehme** [comment]
I think, they'll probably submit on the last day for two reasons: 1) As with every deadline, things get done only right before the deadline, and 2) you don't want to give a heads up to your competitors. Most of the preparations are probably going to happen behind closed doors. Would still be good to have them registered well beforehand.

**Comment by Alan32Liu** [comment]
I see!
Would we need a pre-run before the final due date?
The pre-run will help us to confirm that our fuzzing framework is working well and that we are only one 'button click' away from the official evaluation. The pre-run results will also let the participant know if their fuzzers are running as expected within our framework on the training set.

However, that will require them to submit their work before the official due date and will give them some time to modify their code (fixing errors, optimising performance, etc).

**Comment by mboehme** [comment]
They can pre-run. There is a `make presubmit` functionality which they can run locally and which should cover some of the most critical Github Actions. We should probably also ask them to run a local FuzzBench experiment successfully for 30mins, as well.

**Comment by Alan32Liu** [comment]
OK, I will make a Google Form and share it with you for review by Monday.
The form will collect the information discussed above and allow them to register earlier than the due date.
But I will also prepare to get many PRs at the last minute.

**Comment by mboehme** [comment]
Maybe we can ask them to submit by a certain date and then let them fix failed builds for a week (or something). Wdyt?

**Comment by Alan32Liu** [comment]
Yes! Here is the approximate timeline in my mind (but it may be subject to changes later):
1. T-?(When we are ready): Registration starts. From this point, participants can clone FuzzBench, run local experiments, and submit public PRs to FuzzBench when they see fit. We will accept/reject applications during this period. We may also want to use a replica of FuzzBench dedicated to this competition, add more private benchmarks to it as the testing set, and freeze the framework & public training set before this date.

2. T-14: Pre-run starts. We will no longer take any new applications from this date and start the pre-run evaluation on all existing submissions with the same configuration as the final evaluation, but only on the training set of benchmarks.

3. T-7: Pre-run results released. Participants will have a week for minor modifications on their PR.

4. T-0: Official due date. We freeze their submissions and start the final evaluation with the complete benchmark set.

5. T+7: Final results published. We publish the results and notify the participants about their final performance and ranking.

Would that be similar to what you have in mind?

**Comment by Alan32Liu** [comment]
I've sent you the form to review : )
Two quick questions:
1. Do we have a logo/background picture I can use as the form's header? The [header background of SBST'22](https://sbst22.github.io/) is pretty nice: I wonder if we have anything this year?
2. Will every participant be able to publish a paper at the conference, or only the top N of them? If not everyone, would it make sense to only ask for an abstract when they register and require the full report at the conference's due date?

I also added a question asking if the participant is an author of the fuzzer or has the authors' consent. That could help us filter out some participants if we like.

**Comment by mboehme** [comment]
My worry with the current setup is that competitors will be required to develop their fuzzer in the open. I think they should be allowed to develop the fuzzer privately, and submit their PRs in the last minute (from which time on they have a few more days to get everything working within the public PR). Concretely this means, I would detach the registration and the PR submission. Wdyt?

**Comment by mboehme** [comment]
> 1. Do we have a logo/background picture I can use as the form's header? The [header background of SBST'22](https://sbst22.github.io/) is pretty nice: I wonder if we have anything this year?

Let's ask this question in the SBFT Slack channel.

> 2. Will every participant be able to publish a paper at the conference, or only the top N of them? If not everyone, would it make sense to only ask for an abstract when they register and require the full report at the conference's due date?

Yes, everyone will be able to publish their 2 page report. An initial draft should be part of the registration and the full 2-page report should be part of the final submission (PR).

**Comment by Alan32Liu** [comment]
> My worry with your proposed setup is that competitors will be required to develop their fuzzer in the open, which makes it difficult to compete, right? People will know the tricks and opportunities that the others use. I think they should be allowed to develop the fuzzer privately, and submit their PRs in the last minute (from which time on they have a few more days to get everything working within the public PR). Concretely this means, I would detach the registration and the PR submission. Wdyt?

Ah, sorry: I reckon this setup allows them to develop their fuzzer privately and push their fuzzer at the last minute.
If they didn't want to reveal their secret, they could create an empty PR as a _placeholder_ when they submit their registration form and only push the commits that contain their fuzzers (or their secret algorithms) to the PR at the last minute.
I think this is similar to your idea, except that this detaches their registration from individual commits in their PR (instead of the whole PR). Please let me know if I misunderstood it : )
The PR URL in the registration form allows us to link their registration with their fuzzers (and identify invalid/duplicated URLs).
But I should clarify this in the registration form: They don't have to submit the complete code when they create a PR, as they are allowed to modify or push their secrets at the last minute.

Having an early PR has multiple benefits:
1. We have a unified public place to answer their questions. If they have troubles or general questions about CI failures, they don't have to add us to their private repos. Everything is recorded and documented in their PR in our repo. When we see repeated questions, we can link them to our answers in another PR. They can also fix their CI failures by looking at others' code.
2. We know the infrastructure for CI tests is not modified and is the same as our evaluation.
3. They have more budget for running CI tests. If I understand correctly, GitHub only gives a limited CI budget to each account. If their private account runs out of budget, they won't be able to run CI tests in their private repo anymore. Running CI tests in our repo won't use their budget.

Please let me know if that sounds good to you, and I am more than happy to ask them to submit their PR at the last minute if you prefer : )
Thanks!

**Comment by Alan32Liu** [comment]
> Yes, everyone will be able to publish their 2 page report. An initial draft should be part of the registration and the full 2-page report should be part of the final submission (PR).

Cool, I will specify in the registration form that the report they submit only needs to be an initial draft.

**Comment by mboehme** [comment]
> they could create an empty PR as a placeholder when they submit their registration form and only push the commits that contain their fuzzers (or their secret algorithms) to the PR at the last minute.

Oh, I get it now. In this case, why not skip the form entirely and make the registration as PR, i.e., submit the initial draft of the 2-page report (describing the competing fuzzer) as a PR? We should make this entire process, and the benefits of this process clear on the website, as well. Wdyt?

**Comment by Alan32Liu** [comment]
> Oh, I get it now. In this case, why not skip the form entirely and make the registration as PR, i.e., submit the initial draft of the 2-page report (describing the competing fuzzer) as a PR? We should make this entire process, and the benefits of this process clear on the website, as well?
> 
> In the PR, they can create a new folder and dump a README with all the form's information. Instead of the 2 page report, they can also just add and update the README.md.
> 
> Wdyt?

That could work too, e.g., I can set a PR template and request participants to follow.
But Google form can simplify our work in the following ways, which I am not sure if PR supports:
1. All responses will be automatically organized into a private [Google Sheet](https://docs.google.com/spreadsheets/d/1i9G8K1mB1nIQX1Djkaj-95fpmYziHfnTI2Jh41h1-Yc/edit?resourcekey=0-rpms9i6t6RdvStgPRXY-2Q#gid=478856815) that is only visible to us to view/count/sort/filter submissions. The sheet allows us to exchange opinions for private discussions, check for invalid submissions, and identify duplications in a single place. The link contains a sample response that I tested.
2. A participant's response will not be accessible to others, protecting their team member's information (emails, names, and affiliations) and the content of their draft paper.
3. All draft papers will be automatically uploaded into a single [Google Drive folder](https://drive.google.com/corp/drive/folders/1sKPzJV-PVC_Bt3o02-4eu0etMjLIer3D_vp6tuhC-Rd2haULF-dlTK9TlDRYiJ1gNdCkH0zQ). We can download/share them easily.

Speaking of submitting draft papers, would there be any chance that the committee is already using some professional tool (e.g. [easychair](https://easychair.org/)) to manage other submissions? In that case, I would prefer to change the last question in the registration form (i.e., upload draft paper) to that (e.g., request a proof of submission to easychair), as those tools allow participants to update their papers at any time and perform automatic sanity checks.

**Comment by mboehme** [comment]
Great points! Google Form and PR sounds good to me. Uploading the report via the registration form is fine with me, too. Adding easychair may be overkill 😆.

**Comment by Alan32Liu** [comment]
Thanks!
I've edited the registration form as we discussed, could you please take any look at it at your convenience?
I left some `(link here)` for us to add links later : )

**Comment by mboehme** [comment]
Okay added the links and made small changes. Please have a look. If you are fine, I'd say the form is ready.

**Comment by mboehme** [comment]
Oh. We still need a deadline for registration, @Alan32Liu. Also, I think we don't actually make them wait for accept/reject, right? Just let them register and go. Wdyt? If so, we can set something like 1st Dec.

**Comment by mboehme** [comment]
Registration: 13th Jan 2023 (see SBFT slack).

**Comment by Alan32Liu** [comment]
> Oh. We still need a deadline for registration, @Alan32Liu. Also, I think we don't actually make them wait for accept/reject, right? Just let them register and go. Wdyt? If so, we can set something like 1st Dec.

Yes! I think that is the easiest way: They will register and go; We will announce all results on 13th Jan, and email their primary contact if we accept (and reject?) their fuzzer.
If we have time, we can also review some registrations as soon as they are submitted, but let's hold the results until Jan 13th.

**Comment by Alan32Liu** [comment]
> Registration: 13th Jan 2023 (see SBFT slack).

I will prepone the deadline to 6 Jan as discussed in the PR, hope you won't mind : )
I reckon this deadline won't affect much, as they only need to express their interests and don't have to have their fuzzer or report ready before this date.

**Comment by Alan32Liu** [comment]
I just edited the registration form as discussed (prepone due date + unify the naming of 'tool report').
Please take a look and let me know if further change is required : )
Thanks!
